### PR TITLE
Added supprt for match bool prefix

### LINKED ISSUE #3780: Add support for match_bool_prefix

### PR DESCRIPTION
**Pull request by fulmicoton**
Closes #3780

### CODE REVIEW

File: docs/reference/es_compatible_api.md [hunk 1]
```diff
@@ -366,7 +366,32 @@ The following query types are supported.
 | `query`           | String     |  Full-text search query. The last token will be prefix-matched   | -       |
 | `zero_terms_query`|  `all` or `none` | Defines if all (`all`) or no documents (`none`) should be returned if the query does not contain any terms after tokenization. | `none` |
 | `max_expansions`  |  `Integer` | Number of terms to be match by the prefix matching.               |  50     |
-| `slop`            | `Integer`  | Allows extra tokens between the query tokens.                    |  0      |
+| `analyzer`        | String     | Analyzer meant to cut the query into terms. It is recommended to NOT use this parameter. |  The actual field tokenizer.  |
```
Comment (fmassot) [inline docs/reference/es_compatible_api.md:369]: I think we don't use the analyzer set in the query in the code. 

See the code in the main branch: 

https://github.com/quickwit-oss/quickwit/blob/main/quickwit/quickwit-query/src/query_ast/full_text_query.rs#L98-L99

And I guess we don't test this.

I'm ok to merge this PR and open issue to fix that or the docs.

File: quickwit/quickwit-query/src/elastic_query_dsl/match_query.rs [hunk 2]
```diff
@@ -46,6 +48,11 @@ pub(crate) struct MatchQueryParams {
     pub(crate) operator: BooleanOperand,
     #[serde(default)]
     pub(crate) zero_terms_query: MatchAllOrNone,
+    /// Currently ignored for Match queries (as we do not handle fuzziness) but used in
+    /// MatchBoolPrefixQuery. In the latter, it will control the maximum number of terms to
+    /// match having prefix the last term of the query.
```
Comment (fmassot) [inline quickwit/quickwit-query/src/elastic_query_dsl/match_query.rs:53]: Not sure if this is clearer but I tried.

```suggestion
    /// Currently ignored for Match queries (as we do not handle fuzziness) but used in
    /// MatchBoolPrefixQuery. In the latter, it limits the maximum number of terms starting with the prefix
    /// defined by the last term of the query.
```

**Review by fmassot — APPROVED** [review]
See comments.

File: docs/reference/es_compatible_api.md [hunk 3]
```diff
@@ -366,7 +366,32 @@ The following query types are supported.
 | `query`           | String     |  Full-text search query. The last token will be prefix-matched   | -       |
 | `zero_terms_query`|  `all` or `none` | Defines if all (`all`) or no documents (`none`) should be returned if the query does not contain any terms after tokenization. | `none` |
 | `max_expansions`  |  `Integer` | Number of terms to be match by the prefix matching.               |  50     |
-| `slop`            | `Integer`  | Allows extra tokens between the query tokens.                    |  0      |
```
[the remark this answers is not in the document]
Comment (fulmicoton) [reply docs/reference/es_compatible_api.md:369]: no that was an error! Thanks for catching.

File: docs/reference/es_compatible_api.md (hunk 1 above)
Comment (fulmicoton) [reply docs/reference/es_compatible_api.md:370]: I think we do use in the body of that method. Let me add a test for this.

Comment (fulmicoton) [reply docs/reference/es_compatible_api.md:370]: I added a shitty test for it. It at least checks that it is not no-op.

**Pull request merged**